### ISSUE TITLE
Build failed with the latest nightly rustc

### ISSUE THREAD

**Issue by zonyitoo** [contain-rs/linked-hash-map#38]
Build log

```
   Compiling linked-hash-map v0.0.7
/home/zonyitoo/.cargo/registry/src/github.com-88ac128001ac3a9a/linked-hash-map-0.0.7/src/lib.rs:112:28: 112:42 error: no associated item named `with_map` found for type `LinkedHashMap<K, V>` in the current scope
/home/zonyitoo/.cargo/registry/src/github.com-88ac128001ac3a9a/linked-hash-map-0.0.7/src/lib.rs:112     pub fn new() -> Self { Self::with_map(HashMap::new()) }
                                                                                                                               ^~~~~~~~~~~~~~
/home/zonyitoo/.cargo/registry/src/github.com-88ac128001ac3a9a/linked-hash-map-0.0.7/src/lib.rs:112:28: 112:42 note: the method `with_map` exists but the following trait bounds were not satisfied: `std::collections::hash::map::RandomState : core::hash::HashState`
/home/zonyitoo/.cargo/registry/src/github.com-88ac128001ac3a9a/linked-hash-map-0.0.7/src/lib.rs:116:9: 116:23 error: no associated item named `with_map` found for type `LinkedHashMap<K, V>` in the current scope
/home/zonyitoo/.cargo/registry/src/github.com-88ac128001ac3a9a/linked-hash-map-0.0.7/src/lib.rs:116         Self::with_map(HashMap::with_capacity(capacity))
                                                                                                            ^~~~~~~~~~~~~~
/home/zonyitoo/.cargo/registry/src/github.com-88ac128001ac3a9a/linked-hash-map-0.0.7/src/lib.rs:116:9: 116:23 note: the method `with_map` exists but the following trait bounds were not satisfied: `std::collections::hash::map::RandomState : core::hash::HashState`
/home/zonyitoo/.cargo/registry/src/github.com-88ac128001ac3a9a/linked-hash-map-0.0.7/src/lib.rs:621:14: 621:18 error: no method named `iter` found for type `&LinkedHashMap<K, V>` in the current scope
/home/zonyitoo/.cargo/registry/src/github.com-88ac128001ac3a9a/linked-hash-map-0.0.7/src/lib.rs:621         self.iter().map(|(k, v)| (k.clone(), v.clone())).collect()
                                                                                                                 ^~~~
/home/zonyitoo/.cargo/registry/src/github.com-88ac128001ac3a9a/linked-hash-map-0.0.7/src/lib.rs:621:14: 621:18 note: the method `iter` exists but the following trait bounds were not satisfied: `std::collections::hash::map::RandomState : core::hash::HashState`, `std::collections::hash::map::RandomState : core::hash::HashState`
/home/zonyitoo/.cargo/registry/src/github.com-88ac128001ac3a9a/linked-hash-map-0.0.7/src/lib.rs:621:14: 621:18 help: items from
�
traits can only be used if the trait is implemented and in scope; the following trait defines an item `iter`, perhaps you need to implement it:
/home/zonyitoo/.cargo/registry/src/github.com-88ac128001ac3a9a/linked-hash-map-0.0.7/src/lib.rs:621:14: 621:18 help: candidate #1: `core::slice::SliceExt`
/home/zonyitoo/.cargo/registry/src/github.com-88ac128001ac3a9a/linked-hash-map-0.0.7/src/lib.rs:621:35: 621:44 error: the type of this value must be known in this context
/home/zonyitoo/.cargo/registry/src/github.com-88ac128001ac3a9a/linked-hash-map-0.0.7/src/lib.rs:621         self.iter().map(|(k, v)| (k.clone(), v.clone())).collect()
                                                                                                                                      ^~~~~~~~~
/home/zonyitoo/.cargo/registry/src/github.com-88ac128001ac3a9a/linked-hash-map-0.0.7/src/lib.rs:621:46: 621:55 error: the type of this value must be known in this context
/home/zonyitoo/.cargo/registry/src/github.com-88ac128001ac3a9a/linked-hash-map-0.0.7/src/lib.rs:621         self.iter().map(|(k, v)| (k.clone(), v.clone())).collect()
                                                                                                                                                 ^~~~~~~~~
```

Building `lru-cache`, which depends on `linked-hash-map v0.0.7`.

**Comment by apasel422** [comment]
I think this is a result of rust-lang/rust#31081. Investigating a fix now.

**Comment by zonyitoo** [comment]
@apasel422 If this works well, please publish it to crates.io. Thanks! :)

**Comment by apasel422** [comment]
This is now published as `0.0.8`.
